### DISCUSSION TITLE
Update XRes and YRes after resize a source file

### DISCUSSION THREAD

**Discussion by ahmadrezarazian** [libvips/libvips#2911] [Q&A]
Hello. Thanks for your library.
I have an issue, after I resize the image by vipsthumbnail.exe and use vipsheader --all to see the meta and EXIF of new file, both of xres and yres are same as input file (source).
I was thinking that after resize these values need to update.
Thank you.

**Comment by jcupitt** [comment]
Hi @ahmadrezarazian,

We've discussed this issue a lot. Our current position is that a library like libvips doesn't have enough information to always manipulate resolution correctly, so it's best to do nothing and leave it to a higher level in the application.

For example, in this case has the user run `resize` because they want to print the image smaller on the page, or because they want to print it at the same size but with less detail? Either is possible.

Summary: resolution changes are impossible to always get right, so we leave it to the user.

**Reply by ahmadrezarazian** [reply]
Dear @jcupitt , Thanks for reply.
Yes I understand.
My problem is after running a command such as "vipsthumbnail.exe input.jpg --size 512x512 -o th.jpg", I cannot trust the XRes and YRes in the new file EXIF, and it is always as the input file.
I am wondred if it is possible to control XRes and YRes by adding A command-line argument for vipsthumbnail.exe or vips.exe such as "--update-xres-yres" or etc. because after resizing the image and save as another format(such as JPEG), there is no command line in vips.exe or vipsthumbnail.exe for this issue.
Thank you.

**Reply by jcupitt** [reply]
You can change the resolution with eg.:

```
vips copy in.jpg out.jpg --xres=12.9 --yres=44.2
```

Where resolution is in pixels per millimetre.

**Reply by ahmadrezarazian** [reply]
Dear @jcupitt , Thanks for the reply.
Your help is great.
Is it possible to modify EXIF (meta data) by vips or vipsheader or vipsthumbnail?
Also, is any manual for vips and vipsthumbnail commands line?
thank you for your time and reply.

**Reply by jcupitt** [reply]
You'll need to use something like python or ruby if you want to edit exif. In python:

```python
image = pyvips.Image.new_from_file("something.jpg")
image = image.copy()
image.set_type(pyvips.GValue.gstr_type, "exif-ifd0-XPComment", "hello there!") 
image.write_to_file("other.jpg")
```

https://www.libvips.org/API/current/Using-vipsthumbnail.html

https://www.libvips.org/API/current/using-cli.html

**Reply by ahmadrezarazian** [reply]
Dear @jcupitt , Thanks.
Is it possible to add --xres and --yres commands in vips.exe resize ? Becasue I checked and these are only for vips.exe copy.
or use xres and yres in vipsthumbnail .exe
Thank you.

**Reply by ahmadrezarazian** [reply]
Dear @jcupitt , Thanks.
Is it possible to add --xres and --yres commands in vips.exe resize ? Becasue I checked and these are only for vips.exe copy.
or possible to use xres and yres in vipsthumbnail .exe
Thank you.

**Reply by jcupitt** [reply]
Sorry @ahmadrezarazian, I think that's out of scope for `vips.exe`. It's only supposed to be a simple thing for testing, it's not a complete image processing command. 

You need to use python / ruby / C# / JS etc. if you want more control.

**Reply by ahmadrezarazian** [reply]
Ok. Thank you.
Can vipsedit edit xres and yres in JPG files? Thanks for your time.

**Reply by jcupitt** [reply]
Sorry, `vipsedit` can only edit vips files (`.v` format).
